include/vcpkg/base/messages.h

```diff
@@ -0,0 +1,203 @@
+    void threadunsafe_initialize_context();
+
+    struct LocalizedString
+    {
+        LocalizedString() = default;
+        operator StringView() const { return m_data; }
+        const std::string& data() const { return m_data; }
+
+        static LocalizedString from_string_unchecked(std::string&& s)
+        {
+            LocalizedString res;
+            res.m_data = std::move(s);
+            return res;
+        }
+
+        LocalizedString& append(const LocalizedString& s);
+
+    private:
+        std::string m_data;
+        // to avoid lock-in on LocalizedString, these are free functions
+        // this allows us to convert to `std::string` in the future without changing All The Code
+        friend LocalizedString& append_newline(LocalizedString&);
+        friend LocalizedString&& append_newline(LocalizedString&& self) { return std::move(append_newline(self)); }
+        friend LocalizedString& appendnl(LocalizedString& self, const LocalizedString& to_append)
+        {
+            return append_newline(self.append(to_append));
+        }
+    };
+
+    inline LocalizedString localized_from_error_code(const std::error_code& ec)
+    {
+        return LocalizedString::from_string_unchecked(ec.message());
+    }
+
+    struct LocalizedStringMapLess
+    {
+        using is_transparent = void;
+        bool operator()(const LocalizedString& lhs, const LocalizedString& rhs) const
+        {
+            return lhs.data() < rhs.data();
+        }
+    };
+
+    void write_unlocalized_text_to_stdout(Color c, StringView sv);
+
+    template<class Message, class... Tags, class... Ts>
+    LocalizedString format(Message, detail::MessageArgument<Tags, Ts>... args)
+    {
+        // avoid generating code, but still typeck
+        // (and avoid unused typedef warnings)
+        static_assert((Message::check_format_args((Tags{})...), true), "");
+        return detail::internal_vformat(Message::index,
+                                        fmt::make_format_args(fmt::arg(Tags::name(), *args.parameter)...));
+    }
+
+    inline void println() { write_unlocalized_text_to_stdout(Color::none, "\n"); }
+
+    inline void print(Color c, const LocalizedString& s) { write_unlocalized_text_to_stdout(c, s); }
+    inline void print(const LocalizedString& s) { write_unlocalized_text_to_stdout(Color::none, s); }
+
+    template<class Message, class... Ts>
+    void print(Message m, Ts... args)
+    {
+        print(format(m, args...));
+    }
+    template<class Message, class... Ts>
+    void println(Message m, Ts... args)
+    {
+        print(append_newline(format(m, args...)));
+    }
+
+    template<class Message, class... Ts>
+    void print(Color c, Message m, Ts... args)
+    {
+        print(c, format(m, args...));
+    }
+    template<class Message, class... Ts>
+    void println(Color c, Message m, Ts... args)
+    {
+        print(c, append_newline(format(m, args...)));
+    }
```

Comment:
I'm planning to remove `system.print.h`; `messages.h` will replace it. For now, this is a useful distinction since it allows me to remove `system.print.h` from a file. We could add a `system.print-messages.h` which include this stuff?

Reply:
In that case can we simply fold `messages.h` into `system.print.h`? Why the rename?